CMakeLists.txt.versioning

```diff
@@ -80,5 +30,65 @@ else()
     list(GET ${PROJECT_NAME}_VERSION_LIST 6 ${PROJECT_NAME}_VERSION_GIT_SHA)
     unset(${PROJECT_NAME}_VERSION_LIST)
 
-    message("-- Loaded AWS IoT Device Client version: ${${PROJECT_NAME}_VERSION_STRING_FULL}")
+    message("-- Failed to infer patch version from git, loaded AWS IoT Device Client version from file: ${${PROJECT_NAME}_VERSION_STRING_FULL}")
+endfunction()
+
+if (GIT_VERSION AND NOT ${GIT_INFO} STREQUAL "")
+    message("-- Using Git to calculate AWS IoT Device Client version information...")
+
+    # Get last tag from git - this only matches tags starting with v, so we ignore non-versioning tags
+    execute_process(COMMAND ${GIT_EXECUTABLE} describe --abbrev=0 --tags --match "v[0-9]*"
```

Comment:
If feel we should check if exit code equal to 0 every time we call execute_process? Then do something like 

```
if (exit EQUAL 1)
    message("Encounter failure when doing XXXXX")
```